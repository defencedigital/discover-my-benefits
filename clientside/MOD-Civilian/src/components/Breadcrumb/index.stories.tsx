import React from 'react';
import { storiesOf } from '@storybook/react';
import { Breadcrumb } from '@components';

storiesOf('Breadcrumb', module).add('Breadcrumb', () => {
  const breadcrumb = [
    {
      text: 'Home',
      slug: `/`,
      active: true,
      icon: 'home',
    },
    {
      text: 'test',
      slug: '/test-page',
      active: false,
    },
  ];
  return (
    <React.Fragment>
      <Breadcrumb items={breadcrumb} />
    </React.Fragment>
  );
});
