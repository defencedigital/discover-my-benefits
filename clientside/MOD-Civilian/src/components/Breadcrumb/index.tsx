import * as React from 'react';
import { BreadcrumbItem, BackButton } from '@components';

interface IBreadcrumbProps {
  items: IBreadcrumbItemProps[];
}

const Breadcrumb = ({ items }: IBreadcrumbProps) => {
  return (
    <React.Fragment>
      <nav className="breadcrumb mt-3 xs:mt-3 py-10 max-w-screen-lg flex flex-wrap lg:flex-no-wrap items-stretch mx-auto px-3 lg:px-0 justify-between">
        <ol className="flex flex-row px-2 bg-primary text-white flex-wrap h-full items-stretch">
          {items.length > 0 &&
            items.map((item, index) => (
              <BreadcrumbItem
                key={`${item.text} ${item.slug ? item.slug : ''}`}
                text={item.text}
                active={!!item.active}
                slug={item.slug ? item.slug : null}
                icon={item.icon ? item.icon : null}
              />
            ))}
        </ol>
        <BackButton text="Back" />
      </nav>
    </React.Fragment>
  );
};

export default Breadcrumb;
