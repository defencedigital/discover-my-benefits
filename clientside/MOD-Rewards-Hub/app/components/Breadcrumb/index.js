import React from 'react';
import { Breadcrumb, BreadcrumbItem } from 'reactstrap';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import GridIcon from '../../images/svg/grid.svg';
import HomeIcon from '../../images/svg/home.svg';

const BC = ({ history, items }) => (
  <Breadcrumb item="a" tag="nav">
    {items.length > 0 &&
      items.map(item => (
        <BreadcrumbItem
          key={`${item.text} ${item.slug ? item.slug : ''}`}
          tag={item.active ? 'a' : 'span'}
          onClick={e => {
            e.preventDefault();
            if (item.active) {
              history.push(item.slug);
            }
          }}
          href={item.active ? item.slug : null}
        >
          {item.icon === 'home' && (
            <span className="breadcrumb-icon breadcrumb-icon--home">
              <HomeIcon />
            </span>
          )}
          {item.icon === 'grid' && (
            <span className="breadcrumb-icon breadcrumb-icon--grid">
              <GridIcon />
            </span>
          )}
          {item.text}
        </BreadcrumbItem>
      ))}
  </Breadcrumb>
);

BC.displayName = 'Breadcrumb';

BC.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      slug: PropTypes.string,
      active: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired,
      icon: PropTypes.string,
    }),
  ).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
};

export default withRouter(BC);
