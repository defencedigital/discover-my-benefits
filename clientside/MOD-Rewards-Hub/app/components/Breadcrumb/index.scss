.breadcrumb{
  margin: 2rem 0 0;
  background-color: $gray-lighter;
  padding: 10px;
  @include media-breakpoint-down(md) {
    padding: 5px;
    margin: 10px 0 0;
  }
  a {
    font-weight: 600;
  }
}

.breadcrumb-item{
  font-weight: 400;
  color: $text-color;
  font-size: 1.6rem;
  line-height: 2.6rem;
  margin-right: 15px;
  display: inline-block;

  @include media-breakpoint-down(md) {
    font-size: 1.2rem;
  }

}

span.breadcrumb-item{

  font-weight: 400;

}

.breadcrumb-icon {
  width: 18px;
  height: 18px;
  display: inline-block;
  margin-right: 15px;
}

.breadcrumb-icon.breadcrumb-icon--grid {
  width: 15px;
  height: 15px;
  margin: 0 15px;
}

.breadcrumb-item +
.breadcrumb-item::before {
  text-decoration: none;
}

.breadcrumb-item+.breadcrumb-item{
  padding-left: 0;
}

