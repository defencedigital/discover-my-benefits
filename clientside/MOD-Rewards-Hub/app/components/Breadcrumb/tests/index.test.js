import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Breadcrumb from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('Breadcrumb component', () => {
  it('matches the snapshot', () => {
    const html = shallow(
      <Breadcrumb
        items={[
          {
            text: 'test',
            slug: '/',
            active: true,
          },
          {
            text: 'test 2',
            slug: '/library',
            active: true,
          },
          {
            text: 'Boostrap',
            active: false,
          },
        ]}
      />,
    );
    expect(html).toMatchSnapshot();
  });
});
