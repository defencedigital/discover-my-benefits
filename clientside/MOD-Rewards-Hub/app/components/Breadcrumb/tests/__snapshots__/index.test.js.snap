// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component matches the snapshot 1`] = `ShallowWrapper {}`;
